cmd/demo/node_setup.go

```diff
@@ -118,7 +121,15 @@ func (n *node) setupContracts() (err error) {
 		if err != nil {
 			return err
 		}
+
 		adjAddr = ewallet.AsEthAddr(tmpAdj)
+
+		ctx, cancel := context.WithTimeout(context.Background(), config.Chain.TxTimeout)
+		defer cancel()
+		err = echannel.ValidateAdjudicator(ctx, n.cb, adjAddr)
+		if err != nil {
+			return err
+		}
```

Comment:
We always use the schema `if err := foo(); err != nil {... }` if we don't need the defined variable(s) outside the `if` block. If we're not lazy we also wrap the error using `errors.WithMessage` (if it already has a stack) or `errors.Wrap` (if it doesn't). It is described very briefly here https://git.perun.network/perun/go-perun/-/wikis/Coding-Code-of-Conduct/General#external-errors-without-a-stack-trace

```suggestion
		if err := echannel.ValidateAdjudicator(ctx, n.cb, adjAddr); err != nil {
			return errors.WithMessage(err, "validating adjudicator contract")
		}
```

Reply:
suggested changes have been adopted to the codebase